index.js

```diff
@@ -416,9 +416,22 @@ SendStream.prototype.pipe = function(res){
   var parts
   if (root !== null) {
     // join / normalize from optional root dir
-    path = normalize(join(root, path))
+    var rawPath = path
+    path = join(root, path)
     root = normalize(root + sep)
 
+    // check for exiting current dir
+    var traversalCheck = rawPath.split(sep);
```

Comment:
I think _eventually in the far distant future_ we'll finally have `path.split` in Node.js core, haha.